system/web/services/RequestService.cfc

```diff
@@ -54,6 +54,15 @@ component extends="coldbox.system.web.services.BaseService"{
 		if( isDefined( "FORM" ) ){ structAppend( rc, FORM ); }
 		if( isDefined( "URL" )  ){ structAppend( rc, URL ); }
 
+		// If the inbound content body is a JSON payload capture it
+		if( 
+			len( context.getHTTPContent() ) 
+			&& isJSON(  context.getHTTPContent() )
+			&& isStruct( context.getHTTPContent( json=true ) ) 
```

Comment:
Only issue is that you end up deserializing twice so you will pay a price for it.

Reply:
Maybe we can apply a regex to the string

Reply:
https://www.regextester.com/95560

Reply:
Yeah, I didn't like that either, but I wasn't sure there was a way around it. It _should_ be valid JSON by the time it passes the `isJSON()` test, but if someone throws an array of objects against it, the regex above is going to match - requiring another conditional to test if the base object is an array.  Either that, or we deserialize it once and nest the conditional for the struct test.